src/main/java/com/phonebookservice/controller/AbstractController.java

```diff
@@ -0,0 +1,33 @@
+package main.java.com.phonebookservice.controller;
+
+import main.java.com.phonebookservice.model.Model;
+import main.java.com.phonebookservice.server.DBIf;
+
+public abstract class AbstractController {
+    private DBIf databse = null;
+
+    /**
+     * controller constructor.
+     *
+     * @param databseValue the database
+     */
+    public AbstractController(final DBIf databseValue) {
+        this.databse = databseValue;
+    }
+
+    /**
+     * create model.
+     *
+     * @param model the model
+     */
+    public abstract void create(Model model);
+
+    /**
+     * get database.
+     *
+     * @return database.
+     */
+    public DBIf getDatabase() {
```

Comment:
Why do we need this get ? I don't think no body care about database inside controller except controller itself. 

Don't add getter to anything except you know that you will need it